src/tools/rbd_mirror/Replayer.cc

```diff
@@ -93,21 +93,29 @@ int Replayer::init()
 void Replayer::run()
 {
   while (!m_stopping.read()) {
-    Mutex::Locker l(m_lock);
     set_sources(m_pool_watcher->get_images());
-    m_cond.WaitInterval(g_ceph_context, m_lock, seconds(30));
+    //m_cond.WaitInterval(g_ceph_context, m_lock, seconds(30));
```

Comment:
you might use something like "rbd_mirror: make Replayer::set_sources() thread-safe" as a title for the commit instead